torch/_higher_order_ops/cond.py

```diff
@@ -130,10 +128,10 @@ def _validate_input(pred, true_fn, false_fn, operands):
             raise RuntimeError("Expect both branches to be callbale.")
 
         if not isinstance(operands, (tuple, list)) or any(
-            not isinstance(t, torch.Tensor) for t in operands
+            not isinstance(t, torch.Tensor) for t in pytree.tree_flatten(operands)[0]
```

Comment:
pytree.tree_flatten(operands)[0] -> pytree.tree_leaves(operands)

Reply:
yes, i can change to tree_any!

> Also is the not isinstance(operands, (tuple, list)) part of the condition still valid?

Yeah, that's one point I want to discuss:
- JAX's cond's signature is `cond(pred, true_fn, false_fn, *operands)`, where operands are positional-only arguments. 
- What we have now is `cond(pred, true_fn, false_fn, operands)`, where operands is a tuple of tensors. Our approach is a little more vobose because users need to wrap inputs with `()` but it looks more structured. 
- The TF approach is `cond(pred, true_fn, false_fn)`, where true_fn and false_fn takes no argument and solely relies on closure.

I have no strong opinion on which one to use. If we'd like to change the signature to be consistent with JAX's or TF's, I can do it in the follow-up pr just that there are quite a number of tests we need to change (i.e. unrolling the input tuple).

Reply:
We should just do ours. We went through an exercise a while ago and determined that we should do (operands) to be consistent with how other control flow op APIs should look like, right?

Reply:
yeah! i recall our discussion.